pandas/core/arrays/arrow/array.py

```diff
@@ -889,9 +889,13 @@ def to_numpy(
             result = np.empty(len(self), dtype=object)
             mask = ~self.isna()
             result[mask] = np.asarray(self[mask]._data)
+        elif self._hasna:
+            data = self.copy()
+            data[self.isna()] = na_value
```

Comment:
So we can't necessarily do `self._data.fill_null(na_value)`?

Reply:
We'd have to check if the na_value is compatible with the arrow array. This raises if not, right now we fall back to object. Is there a function for this in arrow?

Reply:
That's what I thought. Yeah there isn't one AFAICT so this makes sense.

Reply:
We could try except it? Not really elegant :)

Would cut runtime in half again, so would be a nice gain